### DISCUSSION TITLE
What subnet mask should you use in "IP Addresses" section?

### DISCUSSION THREAD

**Discussion by settlej** [netbox-community/netbox#11593] [Q&A] [answered]
When you add a new IP Address do you include the subnet mask where that IP address resides?  Like my computer is a 192.168.10.40 and subnet /24.  Will it be 192.168.10.40/24 or 192.168.10.40/32 because its only 1 ip address?

**Comment by candlerb** [comment, marked as answer]
It's supposed to be 192.168.10.40/24.

**Reply by settlej** [reply]
Thanks for the reply!  Quick question why do we need to specify the mask if the parent prefix already has the subnet mask?

**Reply by jeremystretch** [reply]
https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-does-netbox-require-every-ip-to-have-a-mask

**Reply by candlerb** [reply]
Search the google group and github issues archives.  I think it was down to something to do with query efficiency when configuring an interface, and personally I disagree strongly with it (because for me, data integrity trumps optimisation).

It's a big pain point for me - e.g. with public IPs in a /26 subnet, since users are forced to enter a mask, they often enter /24 as that's what they see more commonly with private IPs.

You can run a report periodically to highlight these inconsistencies:
https://github.com/netbox-community/reports/blob/master/reports/ipam-reports/ip-check-prefix.py

There has been talk (#7845) about the Netbox data model being changed to have an explicit foreign-key link from IP address to prefix, in which case it's possible this could change. However, that's been open for over a year; and if it's changed now it would have big knock-on effects to people who have already integrated systems with Netbox.
